doc/smart-switch/PA-Validation/SmartSwitchPAValidationOffload.md

```diff
@@ -0,0 +1,172 @@
+
+<img src="images/DashOffloadManagerWithConsumer.svg">
+
+The Dash Offload Manager is disabled by default and only enabled for specific platforms that require its functionality.
+
+### 2.1.1 SONiC GNMI server changes
+A new parameter `zmq_dpu_proxy_address_base` is added to the telemetry.go to enable the GNMI Server -> DASH Offload Manager ZMQ connection. If specified, the GNMI server will use `zmq_dpu_proxy_address_base + dpu_index:zmq_port` as a destination for DPU ZMQ connection. For example, on the system with 2 DPUs and base address 127.0.0.10, the DASH Offload Manager's ZMQ server runs on 127.0.0.10:8100 for DPU0 and on 127.0.0.11:8100 for DPU1.
+
+## 2.2 PA Validation Offload
+
+A new orchagent (PaValidationOffloadOrch) is responsible for offloading the PA_VALIDATION table entries. Once started by DashOffloadManager, it subscribes to the PA_VALIDATION table and maps the PA validation entries into the ACL configuration.
+
+### 2.2.1 PA Validation Offload ACL configuration
+
+The offloading is achieved by using a User-Defined EGRESS ACL table bound to the NPU<->DPU port. The table is created/managed per DPU.
+
+```
+{
+    "ACL_TABLE_TYPE": {
+        "DASH_PA_VALIDATION": {
+            "MATCHES": [
+                "TUNNEL_VNI",
+                "SRC_IP",
+                "SRC_IPV6"
+            ],
+            "ACTIONS": [
+                "PACKET_ACTION"
+            ],
+            "BIND_POINTS": [
+                "PORT"
+            ]
+        }
+    }
+}
+```
+
+Please refer to https://github.com/sonic-net/SONiC/blob/master/doc/acl/ACL-Table-Type-HLD.md for more details regarding User-Defined ACL tables.
+
+The ACL table configuration:
+
+```
+{
+  "ACL_TABLE": {
+    "DASH_PA_VALIDATION_DPU0": {
+      "STAGE": "EGRESS",
+      "TYPE": "DASH_PA_VALIDATION",
+      "PORTS": ["Ethernet0"]
+      }
+    }
+}
+```
+
+Each PA Validation entry is translated into the following set of rules:
+
+1) For every address in pa_validation.addresses:
+```
+{
+  "ACL_RULE": {
+    "DASH_PA_VALIDATION_DPU0|RULE_VNI_{pa_validation.vni}_{idx}": {
+      "PRIORITY": "10",
+      "PACKET_ACTION": "ACCEPT",
+      "SRC_IP": "{address}/32",
+      "TUNNEL_VNI": {pa_validation.vni}
+    }
+  }
+}
+```
+
+2) Single drop rule with the lower priority:
+```
+{
+  "ACL_RULE": {
+    "DASH_PA_VALIDATION_DPU0|RULE_VNI_{pa_validation.vni}_DROP": {
+      "PRIORITY": "1",
+      "PACKET_ACTION": "DROP",
+      "TUNNEL_VNI": {pa_validation.vni}
+    }
+  }
+}
+```
+
+<img src="images/DashOffloadAcl.svg">
+
+### 2.2.1 PA Validation Offload GNMI feedback
+To preserve the GNMI feedback behavior, the offload logic must also create an entry in the DPU's APPL_STATE_DB.
+For each PA validation processed, the PaValidationOffloadOrch creates the following entry:
+
+```
+DASH_PA_VALIDATION_TABLE:{{vni}}

```

Comment:
does this mean the NPU side PAValidationOffloadOrch will write into the DPU database?